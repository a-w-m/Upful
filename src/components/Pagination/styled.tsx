import styled from "styled-components"

export const Container = styled.div`
  display: grid;
  grid-template-columns: auto auto auto;
  text-align: center;
  grid-area: auto/1/auto/-1;
  grid-template-areas: "prev num next";
  padding: 2rem 1rem;

  & > a:first-child {
    grid-area: prev;
    font-size: 1.4rem;
    justify-self: start;
    color: var(--secondary-color);
    text-decoration: none;
    transition: all 0.2s ease;
    justify-self: end;
  }
  & > p {
    font-family: var(--main-font);
    font-size: 1.4rem;
    grid-area: num;
  }

  & > a:last-child {
    grid-area: next;
    font-size: 1.4rem;
    justify-self: end;
    color: var(--secondary-color);
    text-decoration: none;
    transition: all 0.2s ease;
    justify-self: start;
  }

  & > a:hover {
    color: var(--pagination-link-hover);
  }
`
