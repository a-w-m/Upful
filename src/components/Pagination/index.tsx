import React from "react"
import { Link } from "gatsby"
import { Container } from "./styled"

interface Props {
  currentPage: number
  numPages: number
  category: string
}

const Pagination: React.FC<Props> = props => {
  const { currentPage, numPages, category } = props
  const prevPage = currentPage - 1
  const nextPage = currentPage + 1

  return (
    <Container>
      {prevPage > 0 && (
        <Link
          to={prevPage <= 1 ? `/${category}/` : `/${category}/${prevPage}/`}
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="14"
            height="14"
            fill="currentColor"
            className="bi bi-chevron-left"
            viewBox="0 0 16 16"
          >
            <path
              fillRule ="evenodd"
              d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
            />
          </svg>
        </Link>
      )}
      <p>
        Page {currentPage} of {numPages}
      </p>
      {nextPage <= numPages && (
        <Link
          to={
            nextPage > numPages
              ? `/${category}/${currentPage}`
              : `/${category}/${nextPage}/`
          }
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="14"
            height="14"
            fill="currentColor"
            className="bi bi-chevron-right"
            viewBox="0 0 16 16"
          >
            <path
              fillRule="evenodd"
              d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
            />
          </svg>
        </Link>
      )}
    </Container>
  )
}

export default Pagination
