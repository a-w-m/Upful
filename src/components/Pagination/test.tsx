import React from "react"
import { render, screen } from "@testing-library/react"
import Pagination from "../Pagination"

const mockData = {
  numPages: 3,
  currentPage: 1,
  category: "TEST",
}

describe("Pagination", () => {
  it("matches snapshot", () => {
    const { container } = render(<Pagination {...mockData} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("should display the current page", () => {
    render(<Pagination {...mockData} />)
    const pageInfo = screen.getByText(
      `Page ${mockData.currentPage} of ${mockData.numPages}`
    )
    expect(pageInfo).toBeInTheDocument()
  })

  it("should not display previous page link when on first page", () => {
    render(<Pagination {...mockData} />)
    const previousLink = screen.queryByRole("link", { name: /previous/i })
    expect(previousLink).toBeNull()
  })

  it("should link to category page when the previous page is the first page", () => {
    render(<Pagination {...mockData} currentPage={mockData.currentPage+1} />)
    const previous = screen.getByRole("link", { name: /previous/i })
    expect(previous).toHaveAttribute('href', `/${mockData.category}/`)
  })

  it("should link to previous numbered page when the previous page is not the first page", () => {
    render(<Pagination {...mockData} currentPage={mockData.numPages} />)
    const previous = screen.getByRole("link", { name: /previous/i })
    expect(previous).toHaveAttribute('href', `/${mockData.category}/${mockData.numPages-1}/`)
  })

  it("should not display next page link when on last page", () => {
    render(<Pagination {...mockData} currentPage={mockData.numPages} />)
    const next = screen.queryByRole("link", {name: /next/i})
    expect(next).toBeNull()
  })

  it("should link to the next page when not on last page", ()=>{
    render(<Pagination {...mockData} />)
    const next = screen.queryByRole("link", {name: /next/i})
    expect(next).toHaveAttribute('href', `/${mockData.category}/${mockData.currentPage+1}/`)

  })
})
