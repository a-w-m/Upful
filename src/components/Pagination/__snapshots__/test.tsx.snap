// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component matches snapshot 1`] = `
<div
  class="sc-bdfBwQ cgDhjj"
>
  <p>
    Page 
    1
     of 
    3
  </p>
  <a
    href="/TEST/2/"
  >
    <svg
      class="bi bi-chevron-right"
      fill="currentColor"
      height="14"
      viewBox="0 0 16 16"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
        fill-rule="evenodd"
      />
    </svg>
  </a>
</div>
`;
